lib/fail.js

```diff
@@ -34,32 +34,39 @@ export default async (pluginConfig, context) => {
     const openFailTitleIssues = await got(openFailTitleIssueEndpoint, { ...apiOptions }).json();
     const existingIssue = openFailTitleIssues.find((openFailTitleIssue) => openFailTitleIssue.title === failTitle);
 
-    if (existingIssue) {
-      debug("comment on issue: %O", existingIssue);
+    const canCommentOnOrCreateIssue = failCommentCondition
```

Comment:
With this setup we will always have some additional API calls even if we do not want to create an or comment on an issue. Setting `failTitle` or `failComment` to `false` had the advantage of an early stop. This is - currently - not possible and I have no idea if it is possible with `failCommentCondition`.

Reply:
Do I understand you correct, that your suggestion is, to call 
```ts
const canCommentOnOrCreateIssue = failCommentCondition
  ? template(failCommentCondition)({ ...context })
  : true;
```
also before
```ts
  if (failComment === false || failTitle === false) {
```
and add
```ts
  if (failComment === false || failTitle === false || canCommentOnOrCreateIssue  === false) {
```
or change it to
```ts
  const isFailureReportDisabled = failComment === false ||
    failTitle === false ||
    canCommentOnOrCreateIssue  === false;
  if (isFailureReportDisabled) {
```
?

We would have to call it twice then:
1. first without the potentially existing issue
2. then with the issue if already existing

Same would apply to success handling. Or do I miss something?

I am totally fine with calling it twice to cover every case and also reduce the API calls, I just want to have it discussed as I am not sure with the best approach.

Reply:
@JonasSchubert Evaluating the template without issue/MR might be problematic depending on the logic that folks bake in there.

I would really only do the optimization for the simplest case where `{success,fail}CommentCondition` is statically set to `false`.